src/box/memtx_tx.c

```diff
@@ -3628,13 +3630,15 @@ memtx_tx_snapshot_cleaner_create(struct memtx_tx_snapshot_cleaner *cleaner,
 		struct memtx_tx_snapshot_cleaner_entry entry;
 		entry.from = tuple;
 		entry.to = clean;
+		entry.in_pk = story->link[0].in_index != NULL;
```

Comment:
We don't actually, since the cleaner is only used for lookup of tuples received from the index. Made the hash table only cotain entries for in-indx tuples. See the [comment below](https://github.com/tarantool/tarantool/pull/10478#discussion_r1774917489).